LEAF_Request_Portal/js/formGrid.js

```diff
@@ -45,21 +45,30 @@ var LeafFormGrid = function(containerID, options) {
      * @param values (required) object of cells and names to generate grid
      * @memberOf LeafFormGrid
      */
-    function printTableReportBuilder(values) {
+    function printTableReportBuilder(values, columnValues) {
         var gridBodyBuffer = '';
         var gridHeadBuffer = '';
         var rows = values.cells === undefined ? 0 : values.cells.length;
         var columns = values.format.length;
+        var numOfColumns = columnValues.length;
         var columnOrder = [];
         var delim = '<span class="nodisplay">^;</span>'; // invisible delimiters to help Excel users
         var delimLF = "\r\n";
         var tDelim = '';
 
+        // remove unused columns
+        values.format = values.format.filter(function(value) {
+            return columnValues.includes(value.id);
+        });
+
+
         //finds and displays column names
         for(var i = 0; i < columns; i++){
-            tDelim = (i == columns-1) ? '' : delim;
-            gridHeadBuffer +='<td style="width: 100px;">' + values.format[i].name + tDelim + '</td>';
-            columnOrder.push(values.format[i].id)
+            if (columnValues.includes(values.format[i].id)) {
```

Comment:
Getting error here during testing, 
![image](https://user-images.githubusercontent.com/17956374/132704336-c624ce1e-cd0c-4943-8d55-c8e955859530.png)

Reply:
I got the same.  I sent him an image of the checkbox conditions that seemed to be associated with it.

Reply:
This looks to be working now